mmpose/engine/hooks/visualization_hook.py

```diff
@@ -103,54 +101,59 @@ def after_val_iter(self, runner: Runner, batch_idx: int,
         pred_sample = outputs[0]
         if total_curr_iter % self.interval == 0:
             self._visualizer.add_datasample(
-                osp.basename(img_path) if self.show else 'val_img',
+                os.path.basename(img_path) if self.show else 'val_img',
                 img,
                 data_sample=pred_sample,
+                draw_gt=False,
                 show=self.show,
                 wait_time=self.wait_time,
                 kpt_score_thr=self.score_thr,
                 step=total_curr_iter)
 
-    def after_test_iter(self, runner: Runner, batch_idx: int,
-                        data_batch: Sequence[dict],
+    def after_test_iter(self, runner: Runner, batch_idx: int, data_batch: dict,
                         outputs: Sequence[PoseDataSample]) -> None:
         """Run after every testing iterations.
 
         Args:
             runner (:obj:`Runner`): The runner of the testing process.
             batch_idx (int): The index of the current batch in the val loop.
```

Comment:
I think line 97 should be `data_samples`, and `val` -> `test`.

Reply:
Line 97 should be `outputs`, otherwise a `TypeError` will be raised. 
`val` has been changed to `test` in line 119